src/components/internals/localized-input-toggle/localized-input-toggle.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import FlatButton from '@commercetools-uikit/flat-button';
+import { WorldIcon } from '@commercetools-uikit/icons';
+import messages from '../messages/localized-input/messages';
+
+const LocalizedInputToggle = (props) => {
+  const intl = useIntl();
+
+  const labelMessage = props.isOpen ? props.hideMessage : props.showMessage;
+  const label =
+    typeof labelMessage === 'string'
+      ? labelMessage
+      : intl.formatMessage(labelMessage, {
+          remainingLanguages: props.remainingLocalizations,
+        });
+
+  return (
+    <FlatButton
+      icon={props.icon}
+      label={label}
+      onClick={props.onClick}
+      isDisabled={props.isDisabled}
+    />
+  );
+};
+
+const intlMessageShape = PropTypes.shape({
```

Comment:
I think `react-intl` exposes a prop-type shape
https://github.com/formatjs/formatjs/blob/8a206d0c2c5d519cc40baa3435e4681f43e12be8/website/docs/guides/distribute-libraries.md

Reply:
Yes but didn't we say we don't want to use shared prop-types shapes as ESLint won't be able to lint them and as we wan't to be "clear" about the part of the shape we need/use in the component?

Reply:
> Yes but didn't we say we don't want to use shared prop-types shapes as ESLint

honestly, I don't recall. discard my comment